diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt

```diff
@@ -0,0 +1,260 @@
+                    if(elseBlock.hasChildOfType(BLOCK)) {
+                        elseBlock.getFirstChildWithType(BLOCK)!!.addChild(elseKeyWord.treePrev.treePrev,
+                                elseBlock.getFirstChildWithType(BLOCK)!!.firstChildNode.treeNext)
+                        elseBlock.getFirstChildWithType(BLOCK)!!.addChild(PsiWhiteSpaceImpl("\n"),
+                                elseBlock.getFirstChildWithType(BLOCK)!!.firstChildNode.treeNext)
+                        node.removeChild(elseKeyWord.treePrev.treePrev)
+                    } else {
+                        val text = "else { \n${elseBlock.treePrev.treePrev.text}\n ${elseBlock.text} \n }"
+                        node.removeChild(elseBlock)
+                        node.addChild(KotlinParser().createNode(text), null)
+                    }
+
+                }
+            }
+        }
+    }
+
+    private fun checkCommentsInCodeBlocks(node: ASTNode) {
+        if (isFirstComment(node)) {
+            if (node.treeParent.elementType == BLOCK || node.treeParent.elementType == CLASS_BODY)
+                checkFirstCommentSpaces(node)
+            else
+                checkFirstCommentSpaces(node.treeParent) // TreeParent is property. Then check white spaces before property
+            return
+        }
+
+        if (!node.treePrev.isWhiteSpace()) {
+            WRONG_NEWLINES_AROUND_KDOC.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                node.treeParent.addChild(PsiWhiteSpaceImpl("\n"), node.treeParent) // If treeParent is property
+            }
+        } else {
+            if (node.treePrev.numNewLines() == 1 || node.treePrev.numNewLines() > 2) {
+                WRONG_NEWLINES_AROUND_KDOC.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                    (node.treePrev as LeafPsiElement).replaceWithText("\n\n")
+                }
+            }
+        }
+    }
+
+    private fun checkSpaceBetweenPropertyAndComment(node: ASTNode, configuration: KdocCodeBlocksFormattingConfiguration) {
+        if (node.treeParent.elementType == PROPERTY
+                && node.treeParent.firstChildNode != node) {
+            if (!node.treePrev.isWhiteSpace()) {
+                COMMENT_WHITE_SPACE.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                    node.treeParent.addChild(PsiWhiteSpaceImpl(" ".repeat(configuration.maxSpaces)), node)
+                }
+            } else if (node.treePrev.text.length != configuration.maxSpaces) {
+                COMMENT_WHITE_SPACE.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                    (node.treePrev as LeafPsiElement).replaceWithText(" ".repeat(configuration.maxSpaces))
+                }
+            }
+        }
+    }
+
+    private fun checkWhiteSpaceBeforeComment(node: ASTNode) {
+        if (node.text.trim('/', '*').takeWhile { it == ' ' }.length == APPROPRIATE_COMMENT_SPACES)
```

Comment:
For block comments having newline after `/*` you shouldn't count spaces:
```
/*
 comment
*/
```
will trigger warning in your code